### PR TITLE
Calculator: Comma in Floating Point Numbers

### LINKED ISSUE #4156: Calculator Suggestion: Support comma in floating numbers
Calculator should support comma as decimals separator in floating numbers.

Example:
ddg:  2,2 - 3 = 19
google 2,2 - 3 = -0.8

------
IA Page:  http://duck.co/ia/view/calculator 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @pjhampton

### PR DESCRIPTION
**Pull request by BillyBarbaro**
## Description of new Instant Answer, or changes
Uses the detected style to substitute in a . for the floating point separator and , as the thousands separator in submitted queries.  This PR also modifies the expected value of some tests affected by this change.  I did my best to keep the test consistent with the desired behavior of the Goodie.  Please let me know if there are any I misinterpreted.

## Related Issues and Discussions
Fixes #4156 

## People to notify
@pjhampton 
@samsk

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Calculator.t [hunk 1]
```diff
@@ -149,10 +149,10 @@ ddg_goodie_test(
         '0.8158 - 0.8157'
     ),
     '2,90 + 4,6' => build_test(
-        '2,90 + 4,6'
+        '2.90 + 4.6'
     ),
     '2,90 + sec(4,6)' => build_test(
```
Comment (pjhampton) [inline t/Calculator.t]: This looks awesome so far 🤘 😄 🤘 
I hate to nit pick at tests, but can we add more complex examples such as `3,44 + 8.44 - 3,1`

Comment (BillyBarbaro) [inline t/Calculator.t]: Just to confirm, the test you gave above should fail since it's using mixed notation?

Comment (pjhampton) [inline t/Calculator.t]: Ok, so you're failing on mix notation? I thought you weren't, but that is cool 👍

File: lib/DDG/Goodie/Calculator.pm [hunk 2]
```diff
@@ -102,6 +103,19 @@ sub rewriteQuery {
     return $text;
 }
 
+# Based on the style detected by NumberStyler
+# Makes sure . is used as the decimal sign
```
Comment (mintsoft) [inline lib/DDG/Goodie/Calculator.pm]: The `NumberStyler` already has this implemented as `for_computation`; you can use that instead of re-implementing

**Comment by pjhampton** [comment]
I think we will expediate this as there is complaints coming in regarding the lack of this feature. Let's aim to merge this tomorrow 👍

**Comment by pjhampton** [comment]
Just a heads up, I'm hoping this will be merged and shipped by Monday.
@BillyBarbaro if you would also use mintsoft's solution, that would be awesome!
Thanks for all your hard work! 💯  💛

**Comment by BillyBarbaro** [comment]
@pjhampton I had a pretty busy last week.  Just substituting in the line @mintsoft suggested for what I wrote causes a bunch of the tests to fail since I think it remove some of the whitespace.  I should have time to take a look into this tomorrow if a one day delay is alright, but I understand if you need to give this to someone else.

**Comment by BillyBarbaro** [comment]
@pjhampton and @mintsoft  Quick question on using the `for_computation` call.  That function removes all the thousands separators, e.g. 1,234,567 becomes 1234567.  Is this ok?

**Comment by pjhampton** [comment]
@BillyBarbaro yeah, that's fine. It's reworked in the UI to display commified in the ledger 👍

**Comment by BillyBarbaro** [comment]
@pjhamption Made the requested change.  All appears to be working.  Let me know what you think.

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**